src/BayesFilters/include/BayesFilters/utils.h

```diff
@@ -61,6 +61,28 @@ std::unique_ptr<T> make_unique(Args&& ...args)
 double log_sum_exp(const Eigen::Ref<const Eigen::VectorXd>& arguments);
 
 
+/**
+ * Evaluate a multivariate Gaussian probability density function.
+ *
+ * @param input Input representing the argument of the function as a vector or matrix.
+ * @param mean The mean of the associated Gaussian distribution as a vector.
+ * @param covariance The covariance matrix of the associated Gaussian distribution as a matrix.
+ *
+ * @return The value of the density function evaluated on the input data as a vector.
+ */
+template<typename Derived>
+Eigen::VectorXd multivariate_gaussian_density(const Eigen::MatrixBase<Derived>& input, const Eigen::Ref<const Eigen::VectorXd>& mean, const Eigen::Ref<const Eigen::MatrixXd>& covariance)
+{
+    const auto diff = input.colwise() - mean;
+
+    Eigen::VectorXd values(diff.cols());
+    for (std::size_t i = 0; i < diff.cols(); i++)
+        values(i) = std::exp(- 0.5 * (static_cast<double>(diff.rows()) * std::log(2.0 * M_PI) + std::log(covariance.determinant()) + (diff.col(i).transpose() * covariance.inverse() * diff.col(i))));
```

Comment:
I would split this function in two functions.
1. `multivariate_gaussian_density` that evaluates the classical multivariate Gaussian density without doing the exp of the log of some of its parts.
2. `multivariate_gaussian_log_density` that evaluates the log likelihood, which is convenient in many scenarios to avoid numerical problems.

Reply:
Let me reformulate this.
I suggest to implement `multivariate_gaussian_log_density` and then trivially have `multivariate_gaussian_density` as the exp of the output of `multivariate_gaussian_log_density`. In this way we have a numerically stable density and log-density.

Reply:
I agree on the fact that it would be better to split in two functions in case a user needs the log-likelihood. However, I would then implement `multivariate_gaussian_density` by taking the exponential of the log-likelihood in order to avoid numerical issues that might happen when multiplying the components of the Gaussian density.

What do you think?

Reply:
Ok, then we agree :+1: